rex/utilities/hpc.py

```diff
@@ -0,0 +1,738 @@
+        user : str | None
+            HPC username. None will get your username using getpass.getuser()
+        qformat : str | None
+            Queue format string specification. Changing this form the
+            default (None) could have adverse effects!
+        skip_rows : int | list | None
+            Optional row index values to skip.
+
+        Returns
+        -------
+        stdout : str
+            HPC queue output string. Can be split on line breaks to get list.
+        """
+
+    @property
+    def queue(self):
+        """Get the HPC queue parsed into dict format keyed by integer job id
+
+        Returns
+        -------
+        queue : dict
+            HPC queue parsed into dictionary format keyed by integer job id
+            with nested dictionary of job properties (queue printout columns).
+        """
+        if self._queue is None:
+            qstr = self.query_queue(user=self._user)
+            self._queue = self.parse_queue_str(qstr)
+
+        return self._queue
+
+    @property
+    def queue_job_names(self):
+        """Get a list of the job names in the queue"""
+        return [attrs[self.QCOL_NAME] for attrs in self.queue.values()]
+
+    @property
+    def queue_job_ids(self):
+        """Get a list of the job integer ids in the queue"""
+        return list(self.queue.keys())
+
+    def check_status(self, job_id=None, job_name=None):
+        """Check the status of an HPC job using the HPC queue.
+
+        Parameters
+        ----------
+        job_id : int | None
+            Job integer ID number (preferred input)
+        job_name : str
+            Job name string.
+
+        Returns
+        -------
+        status : str | NoneType
+            Queue job status str or None if not found.
+            SLURM status strings: PD, R, CG (pending, running, complete).
+            PBS status strings: Q, R, C (queued, running, complete).
+        """
+
+        status = None
+
+        if job_id is not None:
+            if int(job_id) in self.queue:
+                status = self.queue[int(job_id)][self.QCOL_STATUS]
+
+        elif job_name is not None:
+            if job_name in self.queue_job_names:
+                for attrs in self.queue.values():
+                    if attrs[self.QCOL_NAME] == job_name:
+                        status = attrs[self.QCOL_STATUS]
+                        break
+
+        else:
+            msg = 'Need a job_id or job_name to check HPC job status!'
+            logger.error(msg)
+            raise HpcError(msg)
+
+        return status
+
+
+class PBS(HpcJobManager):
+    """Subclass for PBS subprocess jobs."""
+
+    # PBS qstat column headers
+    QCOL_NAME = 'Name'  # Job name column
+    QCOL_ID = 'Job id'  # Job integer ID column
+    QCOL_STATUS = 'S'  # Job status column
+
+    # Frozen PBS qstat column headers cached b/c its not space delimited
+    QSTAT_KEYS = ('Job id', 'Name', 'User', 'Time Use', 'S', 'Queue')
```

Comment:
Since we are defining these, and thoughts about dropping spaces?

Reply:
I think we should leave it how we expect PBS to print it. If the user is interacting with a dictionary of the qstat they would expect the qstat column headers, not some arbitrary ones that we pick

Reply:
Also fyi this is just from some example online. I can't remember what PBS actually prints out and i expect its different from PBS config to config.

Reply:
As we said 90% there!